src/ray/core_worker/task_event_buffer.cc

```diff
@@ -137,23 +137,26 @@ Status TaskEventBufferImpl::Start(bool auto_flush) {
     sigaddset(&mask, SIGTERM);
     pthread_sigmask(SIG_BLOCK, &mask, NULL);
 
-    // Decrease the thread priority to allow worker threads to run.
-    int new_nice = std::min(
-        RayConfig::instance().worker_niceness() + kTaskEventBufferAdditionalNice, 19);
-    new_nice = nice(new_nice);
-    if (new_nice == -1) {
-      RAY_LOG(WARNING) << "Failed to set nice(" << new_nice
-                       << ") for task event buffer io thread: " << strerror(errno);
-    } else {
-      RAY_LOG(INFO) << "Current task event io thread's nice = " << new_nice;
-    }
-
 #endif
     SetThreadName("task_event_buffer.io");
     io_service_.run();
     RAY_LOG(INFO) << "Task event buffer io service stopped.";
   });
 
+#ifdef __linux__
+  // Decrease the thread priority to allow worker threads to run.
+  // Skipping it for mac since niceness is applied to all threads on mac.
+  int new_nice = std::min(
+      RayConfig::instance().worker_niceness() + kTaskEventBufferAdditionalNice, 19);
+  new_nice = nice(new_nice);
```

Comment:
this isn't happening in the thread anymore, right?

Reply:
this will only happen on Linux on a per thread basis.